test/test_tensorboard.py

```diff
@@ -59,7 +59,7 @@ def setUp(self):
         self.temp_dirs = []
 
     def createSummaryWriter(self):
-        temp_dir = str(uuid.uuid4())
+        temp_dir = tempfile.TemporaryDirectory(prefix="test_tensorboard").name
```

Comment:
Don't you need to keep the object alive to ensure the directory is not deleted?

Reply:
Not 100% sure, but `SummaryWriter` would create the directory so I think it doesn't matter. The code before the PR doesn't create directories either.

Reply:
Sure, but then how is this better than the previous code? This can still (unlikely) conflict between different runs and never clean up the temporary folders.
The only difference is that the folder is not in the current directory but in the general tmp one. Is that the only benefit we were looking for here?

Reply:
Yes, the problem I had is that PWD is not writable in our build/test system.

Reply:
Ok sounds good. Could you add a comment here to that effect as it can be a bit confusing:
We use this just to get the name of the directory in a writeable place but we don't care about collision or cleaning up.

Reply:
Added some comments to clarify. 
(Didn't mention collision because I think it is too unlikely to worth a mention. tempfile will try 10000 names if there are conflicts )